exasol/analytics/query_handler/query/select.py

```diff
@@ -101,17 +104,19 @@ class ModifyQuery(CustomQuery, AuditData):
     def __init__(
         self,
         query_string: str,
-        # Using DBObjectName instead of str, because counting rows of the modified table
-        # requires to use fully_qualified
-        db_object_name: DBObjectName,
+        db_object_ref: DBObjectNameWithSchema,
         db_object_type: DbObjectType | str,
         db_operation_type: DbOperationType | str,
         audit_fields: dict[str, Any] | None = None,
         audit: bool = False,
     ):
         CustomQuery.__init__(self, query_string)
         AuditData.__init__(self, audit_fields)
-        self.audit_fields[DB_OBJECT_NAME_TAG] = db_object_name
+        self.db_object_ref = db_object_ref
```

Comment:
```suggestion
        self._db_object_ref = db_object_ref
```

Reply:
add getter, this property should be write once similar to the audit_fields which should be append only